content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,255 @@
+
+Try out the code now using @node controller.js@ in the commmand line and you should see in the console that the client sucessfully connects through MQTT.
+
+TODO: IMAGE OF "connected!"
+
+If you have any issues with this, please check out our "usage notes":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter.
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 6 - Send key presses through MQTT
+
+Now that you have established a connection through MQTT to Ably, you can send some data through it. You need to firstly create a function to easily publish messages for the key presses you're interested in. Add the following function to the bottom of @controller.js@:
+
+```[javascript]
+  function publishMessage(channel, message) {
+    client.publish(channel, message, { qos: 1 }, function(err) {
+      if(err) {
+        console.log(err);
+      }
+    }); 
+  }
+``` 
+
+This function simply takes the message you wish to publish, and publishes it through MQTT using @client.publish@ into the channel specified. The setting 'qos' (Quality of Service) represents whether messages are guaranteed to be delivered 'at most once' (0), 'at least once' (1), or 'exactly once' (2). Here you'll use 'at least once', as you want to ensure our inputs are delivered, but also have as little back and forth communication between this device and the server as possible.
+
+Now that you have a method to publish through MQTT, you need to use it to send our key presses. Inside of the function assosciated with the @process.stdin.on('keypress')@, add the following:
+
+```[javascript]
+  if (key && key.ctrl && (key.name == 'c')) {
+    process.stdin.pause();
+    client.end();
+  } else if(key && (key.name == 'left')) {
+  	publishMessage('input', 'left');
+  } else if(key && (key.name == 'right')) {
+  	publishMessage('input', 'right');
+  } else if(key && (key.name == 'up')) {
+  	publishMessage('input', 'up');
+  } else if(key && (key.name == 'down')) {
+  	publishMessage('input', 'down');
+  } else if(key && (key.name == 'return')) {
+  	publishMessage('input', 'return');
+  }
+```
+
+All this will do is use the @publishMessage@ function you created to send the key presses you're interested in. In the case that the user types ctrl-c, the stream will be paused and the client closed so that the program can stop. Otherwise, you're sending the key pressed to the channel @input@.
+
+"See this step in Github":https://github.com/ably/tutorials/TODO
+
+h2. Step 7 - Recieving inputs from Ably
+
+Now that you've set up your controller, it's time to create something to run the game of Snake, and receive the data from Ably that you sent from the controller. You'll be using the normal Ably Realtime protocol for this part. This is possible due to the fact Ably is "Protocol Agnostic":https://support.ably.io/solution/articles/3000054126-do-you-support-mqtt-are-you-protocol-agnostic-, allowing for protocols to be used interchangebly.
```

Comment:
For some reason GitHub markdown shows an additional hyphen after 'Agnostic' due to the link having it